### PR TITLE
Add group by option to make axioms

### PR DESCRIPTION
**Pull request by TeodorescuIoan**
The options can be activated by passing GROUPBYMOD=true to the make axioms command, for example:
```bash
make GROUPBYMOD=true axioms
```

Obs. if group by module option is active the writing will no longer happen in real time. The contents will be dumpted at the end.

### CODE REVIEW

**Review by wkolowski — CHANGES_REQUESTED** [review]
I think you shouldn't change the existing argument order - the new one should come at the end.

Also, can `make` parse arguments by name, instead of positionally? If so, the script should be updated to indicate that.

I also see that you have a huge duplicated chunk of code in both of the `if` branches - this should be abstracted into a function or at least factored out to the front of the `if` somehow.

**Review by palmskog — COMMENTED** [review]
I agree with @wkolowski to put the new option last. Also, recall that in Makefiles you can do stuff like `MY_VAR =? "default"` to always give a value to something, regardless of whether the user provides the option. I think this should be used here.

**Comment by palmskog** [comment]
@wkolowski can you please re-review this? The argument order and other things now look good to me.

File: scripts/axiom-diagnostics.sh [hunk 1]
```diff
@@ -1,12 +1,19 @@
 #!/bin/sh
 
+
+if test -n "$4"
+then
+   GROUPBYFLAG=$4
```
Comment (wkolowski) [inline scripts/axiom-diagnostics.sh:6]: Please use only 2 spaces on indentation to stay consistent with the rest of the file. Also, why is the variable name different here than in the makefile?

```suggestion
  group_by_mod=$4
```

File: scripts/axiom-diagnostics.sh [hunk 2]
```diff
@@ -90,7 +97,10 @@ awk '{if (lastLine=="") {lastLine=$0;next} if ($0 ~ /Closed under the global con
 | \
 sed -r \
 -e '/PrimInt63/d' \
--e '/Uint63/d'
+-e '/Uint63/d' \
+`# If the user passed GROUPBYMOD=true then group the listings by module name.` \
+| \
+if [ "$GROUPBYFLAG" = "true" ]; then sed -E -e 's/\t/|/' -e '/^\|/!s/()\..*$//' | awk 'BEGIN {lastModule=""} {if ($0 ~ /^\|/) { data[lastModule][$0] = 1 } else { lastModule=$0 }} END {for (key in data) { print key; for (axiom in data[key]){ print "\t"axiom; } print "\n"; }}' | sed -E 's/\|//'; else tail -n +1; fi
```
Comment (wkolowski) [inline scripts/axiom-diagnostics.sh:103]:
```suggestion
if [ "$group_by_mod" = "true" ]; then sed -E -e 's/\t/|/' -e '/^\|/!s/()\..*$//' | awk 'BEGIN {lastModule=""} {if ($0 ~ /^\|/) { data[lastModule][$0] = 1 } else { lastModule=$0 }} END {for (key in data) { print key; for (axiom in data[key]){ print "\t"axiom; } print "\n"; }}' | sed -E 's/\|//'; else tail -n +1; fi
```

**Review by wkolowski — APPROVED** [review]
I tested it and it works fine, but please address my suggestions before merging (and make sure it still works after you do all the renames).

**Pull request merged**